### PR TITLE
feat: setup comparator models

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Comment by llorllale** [comment]
Haven't looked too deeply yet but I see these warnings:

```
The swagger spec at "/home/llorllale/dev/trustbloc/edge-service/.build/rest/openapi/spec/comparator/openAPI.yml" showed up some valid but possibly unwanted constructs.
2021/02/04 15:11:20 See warnings below:
2021/02/04 15:11:20 - WARNING: spec has no valid path defined
2021/02/04 15:11:20 - WARNING: response "#/responses/extractionResp" is not used anywhere
2021/02/04 15:11:20 - WARNING: response "#/responses/genericError" is not used anywhere
2021/02/04 15:11:20 - WARNING: response "#/responses/comparisonResp" is not used anywhere
2021/02/04 15:11:20 - WARNING: response "#/responses/configResp" is not used anywhere
2021/02/04 15:11:20 - WARNING: response "#/responses/createAuthorizationResp" is not used anywhere
2021/02/04 15:11:20 - WARNING: definition "#/definitions/Operator" is not used anywhere
```

**Comment by fqutishat** [comment]
> Haven't looked too deeply yet but I see these warnings:
> 
> ```
> The swagger spec at "/home/llorllale/dev/trustbloc/edge-service/.build/rest/openapi/spec/comparator/openAPI.yml" showed up some valid but possibly unwanted constructs.
> 2021/02/04 15:11:20 See warnings below:
> 2021/02/04 15:11:20 - WARNING: spec has no valid path defined
> 2021/02/04 15:11:20 - WARNING: response "#/responses/extractionResp" is not used anywhere
> 2021/02/04 15:11:20 - WARNING: response "#/responses/genericError" is not used anywhere
> 2021/02/04 15:11:20 - WARNING: response "#/responses/comparisonResp" is not used anywhere
> 2021/02/04 15:11:20 - WARNING: response "#/responses/configResp" is not used anywhere
> 2021/02/04 15:11:20 - WARNING: response "#/responses/createAuthorizationResp" is not used anywhere
> 2021/02/04 15:11:20 - WARNING: definition "#/definitions/Operator" is not used anywhere
> ```

@llorllale I will fix them. Are you okay with models

**Comment by llorllale** [comment]
There are many other types being declared in the generated schema that are not in the original:

* AttribyteTypeAndValue
* Certificate
* ExtKeyUsage
* Extension
* IP
* IPMask
* Int
* JSONWebKey
* KeyUsage
* Name
* ObjectIdentifier
* PublicKeyAlgorithm
* SignatureAlgorithm
* URL
* UserINfo

File: pkg/restapi/comparator/operation/models.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	// swagger:allOf Query
+	Query
+	UpstreamAuth struct {
+		EDV string `json:"edv"`
+		KMS string `json:"kms"`
+	} `json:"authTokens"`
+}
+
+// Authorization is given to requesting parties to allow them access to protected resources.
+type Authorization struct {
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/models.go:27]: missing `authToken`

File: pkg/restapi/comparator/operation/models.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	ID              string `json:"id"`
+	RequestingParty string `json:"requestingParty"`
+	Scope           Scope  `json:"scope"`
+}
+
+// Scope is the scope of the authorized access.
+type Scope struct {
+	DocID  string   `json:"docID"`
+	Action []string `json:"action"`
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/models.go:36]: should be `actions`

File: pkg/restapi/comparator/operation/models.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	Caveat Caveat   `json:"caveat"`
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/models.go:37]: should be `caveats`

File: pkg/restapi/comparator/operation/models.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// Caveat are orthogonal, reusable constraints placed on authorization scopes.
+type Caveat struct {
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/models.go:41]: Missing the sub type `ExpiryCaveat`

File: pkg/restapi/comparator/operation/models.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	Type string `json:"type"`
+}
+
+// Operator is an abstract operation.
+type Operator struct {
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/models.go:46]: missing `EqOp`

Comment (fqutishat) [reply pkg/restapi/comparator/operation/models.go:53]: done

File: pkg/restapi/comparator/operation/models.go (hunk 4 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/models.go:42]: done

File: pkg/restapi/comparator/operation/models.go (hunk 3 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/models.go]: done

File: pkg/restapi/comparator/operation/models.go (hunk 2 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/models.go]: done

File: pkg/restapi/comparator/operation/models.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/models.go:27]: done

**Review by soluchok — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

**Review by rolsonquadras — APPROVED** [review]

**Pull request merged**